### PR TITLE
build!: upgrade to expertsystem 0.4.0a1

### LINKED ISSUE #118: Upgrade expertsystem
Reason:
* More conventient handling of particle info
* @Leongrim has some bug

Ideally, we wait until the `expertsystem` is further refactored and has a more stable release, but this is needed for the ongoing trial analyses.

### PR DESCRIPTION
**Pull request by redeboer**
Closes #118

### CODE REVIEW

**Comment by redeboer** [comment]
@Leongrim Have a look whether this helps

**Comment by redeboer** [comment]
This would require setting the minimal requirement to Python 3.7

**Comment by redeboer** [comment]
Probably better to split up this PR:
* one PR that addresses the missing default dynamics
* one PR that addresses #118 (if desirable at all)

**Comment by redeboer** [comment]
@spflueger Let's put this one on hold while the expertsystem is refactored.

File: tests/recipe/test_amplitude_creation.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def test_intensity(self):
             self.particle_list, self.kinematics, self.phsp_sample
         )
         intensity = builder.create_intensity(self.recipe)
-        assert len(intensity.parameters) == 9
+        assert len(intensity.parameters) == 8
```
Comment (spflueger) [inline tests/recipe/test_amplitude_creation.py:69]: why does this change?

Comment (redeboer) [reply tests/recipe/test_amplitude_creation.py:69]: Seems that the meson radii are gone.

In the previous commit on the master (`77f50f1e`), `parameters` is:
image
while in 29f7dc182c93c0f6f60120d87c10669f59f82559 it's:
image

Comment (spflueger) [reply tests/recipe/test_amplitude_creation.py:69]: Ok. We fix that in a separate PR

**Review by spflueger — APPROVED** [review]

**Pull request merged**